src/gui/guiFormSpecMenu.cpp

```diff
@@ -3138,6 +3138,61 @@ void GUIFormSpecMenu::acceptInput(FormspecQuitMode quitmode=quit_mode_no)
 			fields["quit"] = "true";
 		}
 
+		if (current_keys_pending.pending_key_event) {
+			fields["key_event"] = current_keys_pending.key_event.sym();
+
+			#define SET_KEY_MAPPED(m)									\
+				if (current_keys_pending.key_event == getKeySetting(m))	\
+					fields["key_mapped"] = m;
+
+			SET_KEY_MAPPED("keymap_forward");
+			SET_KEY_MAPPED("keymap_backward");
+			SET_KEY_MAPPED("keymap_left");
+			SET_KEY_MAPPED("keymap_right");
+			SET_KEY_MAPPED("keymap_jump");
+			SET_KEY_MAPPED("keymap_special1");
+			SET_KEY_MAPPED("keymap_sneak");
+			SET_KEY_MAPPED("keymap_autoforward");
+			SET_KEY_MAPPED("keymap_drop");
+			SET_KEY_MAPPED("keymap_inventory");
+			SET_KEY_MAPPED("keymap_chat");
+			SET_KEY_MAPPED("keymap_cmd");
+			SET_KEY_MAPPED("keymap_cmd_local");
+			SET_KEY_MAPPED("keymap_console");
+			SET_KEY_MAPPED("keymap_minimap");
+			SET_KEY_MAPPED("keymap_freemove");
+			SET_KEY_MAPPED("keymap_pitchmove");
+			SET_KEY_MAPPED("keymap_fastmove");
+			SET_KEY_MAPPED("keymap_noclip");
+			SET_KEY_MAPPED("keymap_hotbar_previous");
+			SET_KEY_MAPPED("keymap_hotbar_next");
+			SET_KEY_MAPPED("keymap_mute");
+			SET_KEY_MAPPED("keymap_increase_volume");
+			SET_KEY_MAPPED("keymap_decrease_volume");
+			SET_KEY_MAPPED("keymap_cinematic");
+			SET_KEY_MAPPED("keymap_screenshot");
+			SET_KEY_MAPPED("keymap_toggle_hud");
+			SET_KEY_MAPPED("keymap_toggle_chat");
+			SET_KEY_MAPPED("keymap_toggle_fog");
+			SET_KEY_MAPPED("keymap_toggle_update_camera");
+			SET_KEY_MAPPED("keymap_toggle_debug");
+			SET_KEY_MAPPED("keymap_toggle_profiler");
+			SET_KEY_MAPPED("keymap_camera_mode");
+			SET_KEY_MAPPED("keymap_increase_viewing_range_min");
+			SET_KEY_MAPPED("keymap_decrease_viewing_range_min");
+			SET_KEY_MAPPED("keymap_rangeselect");
+			SET_KEY_MAPPED("keymap_zoom");
+			SET_KEY_MAPPED("keymap_quicktune_next");
+			SET_KEY_MAPPED("keymap_quicktune_prev");
+			SET_KEY_MAPPED("keymap_quicktune_inc");
+			SET_KEY_MAPPED("keymap_quicktune_dec");
```

Comment:
Can outdate very quickly. Consider adding a function to `Settings` which returns an array of these `keymap_*` settings.
EDIT: You can also abort as soon one setting matches the key value.

Reply:
Yes, I thought that there should be a function, but I don't see how to get a list of all settings, so it might have to be another hardcoded list.  Aborting is a good idea, I'll do that.

As for EDIT2, these don't *do* anything on their own.  The name is returned when a modder presses one of these keys.